File: src/main/java/com/teamiurg/techexpansion/item/special/BatteryT1.java
package com.teamiurg.techexpansion.item.special;

import java.util.List;

import com.teamiurg.techexpansion.TechExpansion;
import com.teamiurg.techexpansion.item.TEItem;

import cofh.api.energy.ItemEnergyContainer;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.ActionResult;
import net.minecraft.util.EnumActionResult;
import net.minecraft.util.EnumFacing;
import net.minecraft.util.EnumHand;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

public class BatteryT1 extends ItemEnergyContainer {
	
	public BatteryT1() {
		this.setUnlocalizedName("battery_t1").setCreativeTab(TechExpansion.tab_techExpansion).setMaxStackSize(1)
				.setRegistryName("battery_t1");
	}

	public BatteryT1(int capacity) {

		this(capacity, capacity, capacity);
	}

	public BatteryT1(int capacity, int maxTransfer) {

		this(capacity, maxTransfer, maxTransfer);
	}

	public BatteryT1(int capacity, int maxReceive, int maxExtract) {

		this.capacity = capacity;
		this.maxReceive = maxReceive;
		this.maxExtract = maxExtract;
		this.setUnlocalizedName("battery_t1").setCreativeTab(TechExpansion.tab_techExpansion).setMaxStackSize(1)
				.setRegistryName("battery_t1");
	}

	@Override
	public void addInformation(ItemStack stack, EntityPlayer player, List<String> tooltip, boolean idk) {
		tooltip.add("Current energy: " + this.getEnergyStored(stack) + "/" + this.getMaxEnergyStored(stack) + " RF");
	}
	
	@Override
	public ActionResult<ItemStack> onItemRightClick(ItemStack stack, World world,
			EntityPlayer player, EnumHand hand) {
		this.receiveEnergy(stack, 1, false);
		return ActionResult.newResult(EnumActionResult.SUCCESS, stack);
	}
	
	
	
}

File: src/main/scala/com/teamisotope/techexpansion/item/special/BatteryT1.scala
package com.teamisotope.techexpansion.item.special

import java.util

import cofh.api.energy.ItemEnergyContainer
import com.teamisotope.techexpansion.TechExpansion
import net.minecraft.client.renderer.ItemMeshDefinition
import net.minecraft.client.renderer.block.model.{ModelBakery, ModelResourceLocation}
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraftforge.client.model.ModelLoader
import org.apache.commons.math3.util.Precision

class BatteryT1 extends ItemEnergyContainer(1000, 1, 1) {

  this.setRegistryName("battery_t1").setUnlocalizedName("battery_t1").setCreativeTab(TechExpansion.tab_misc).setMaxStackSize(1)

  override def addInformation(stack: ItemStack, playerIn: EntityPlayer, tooltip: util.List[String], advanced: Boolean): Unit = {
    tooltip.add("Energy: " + this.getEnergyStored(stack) + "/" + this.getMaxEnergyStored(stack))
  }

  def initModel() : Unit = {
    val l100: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_l100", "inventory")
    val r100: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r100", "inventory")
    val r200: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r200", "inventory")
    val r300: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r300", "inventory")
    val r400: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r400", "inventory")
    val r500: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r500", "inventory")
    val r600: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r600", "inventory")
    val r700: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r700", "inventory")
    val r800: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r800", "inventory")
    val r900: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r900", "inventory")
    val r1000: ModelResourceLocation = new ModelResourceLocation("techexpansion:misc/battery_t1/battery_t1_r1000", "inventory")

    ModelBakery.registerItemVariants(this, l100, r100, r200, r300, r400, r500, r600, r700, r800, r900, r1000)

    ModelLoader.setCustomMeshDefinition(this, new ItemMeshDefinition() {
      override def getModelLocation(stack: ItemStack): ModelResourceLocation = {

        Precision.round(getEnergyStored(stack), -2) match {
          case 100 => r100;
          case 200 => r200;
          case 300 => r300;
          case 400 => r400;
          case 500 => r500;
          case 600 => r600;
          case 700 => r700;
          case 800 => r800;
          case 900 => r900;
          case 1000 => r1000;
          case _ => l100;
        }
      }
    })
  }

}
